collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -900,10 +900,13 @@ protected void onActivityResult(int requestCode, int resultCode, final Intent in
             case RequestCodes.EX_DECIMAL_CAPTURE:
                 setWidgetData(ExternalAppUtils.getReturnedSingleValue(intent));
                 break;
+            case RequestCodes.MEDIA_FILE_PATH:
+                loadMedia(Uri.fromFile(new File((String) ExternalAppUtils.getReturnedSingleValue(intent))));
```

Comment:
Can't you return `URI` here instead of `absolutePath`? Thanks to that we could merge this case with what we have above for `RequestCodes.IMAGE_CHOOSER` etc.

Reply:
I'm not sure how I feel about that. The `RequestCodes.IMAGE_CHOOSER` requires the returned value to be the intent "data" rather than an extra which means we can't use our `ExternalAppUtils` pattern. I think I'd rather leave `IMAGE_CHOOSER` as the odd one out.

Reply:
We could use `ExternalAppUtils` if we extend it see: https://github.com/getodk/collect/commit/7ab81cf1272a2211de924870aa6ecbd444e7ba98

Reply:
Yeah, I see that. I guess for me, the motivation to have `ExternalAppUtils` is that it provides a way to return data and use returned data without having to care about the details of how that data is "stored" on either side. Introducing a second way of storing/transferring the data means that both sides (the returning and receiving Activity) need to care again. We obviously can't change the camera intent in Android, but I think I'd rather keep it as a special case and have all the Activity objects in our control (and those implementing extensions) use a consistent API. Does that make sense?

Reply:
Yeah I'm ok with it.